bundles/com.espressif.idf.core/src/com/espressif/idf/core/tools/vo/IDFToolSet.java

```diff
@@ -0,0 +1,156 @@
+/*******************************************************************************
+ * Copyright 2024 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.core.tools.vo;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.cdt.cmake.core.ICMakeToolChainFile;
+
+import com.espressif.idf.core.toolchain.ESPToolchain;
+import com.google.gson.annotations.Expose;
+
+/**
+ * VO to hold the idf tools information that is read or exported to the configuration json
+ * 
+ * @author Ali Azam Rana
+ *
+ */
+public class IDFToolSet implements Serializable
```

Comment:
It seems this class can be replaced with Java records

Reply:
@sigmaaa its definitely a good idea but @Expose annotation are not supported in the records as far as I know so it will introduce complexity to restrict exposing certain fields in json export and import